### PR TITLE
feat(Resources/models): (Anomaly Detection-1) Frontend Add anomalydetection resource type 

### PR DESCRIPTION
**Pull request by Noha-ElAbrouki**
Add anomalydetection resource type and show wrench icon if resource of type anomaly detection

### CODE REVIEW

File: www/front_src/src/Resources/translatedLabels.ts [hunk 1]
```diff
@@ -257,3 +257,4 @@ export const labelServiceSeverity = 'Service severity';
 export const labelHostSeverity = 'Host severity';
 export const labelHostSeverityLevel = 'Host severity level';
 export const labelServiceSeverityLevel = 'Service severity level';
+export const labelPerformanceGraphAD = 'Edit performance AD';
```
Comment (Thebarda) [inline www/front_src/src/Resources/translatedLabels.ts]:
```suggestion
export const labelEditAnomalyDetectionData = 'Edit anomaly detection data';
```

File: www/front_src/src/Resources/Graph/Performance/GraphActions.tsx [hunk 2]
```diff
@@ -139,6 +144,21 @@ const GraphActions = ({
           >
             <SaveAsImageIcon style={{ fontSize: 18 }} />
           </IconButton>
+
+          {isResourceAD && (
+            <IconButton
+              disableTouchRipple
```
Comment (Thebarda) [inline www/front_src/src/Resources/Graph/Performance/GraphActions.tsx]:
```suggestion
```

File: www/front_src/src/Resources/Details/tabs/Graph/index.tsx [hunk 3]
```diff
@@ -40,7 +41,9 @@ const GraphTabContent = ({ details }: TabProps): JSX.Element => {
   const equalsService = equals('service');
   const equalsMetaService = equals('metaservice');
 
-  const isService = or(equalsService(type), equalsMetaService(type));
+  const isService =
+    or(equalsService(type), equalsMetaService(type)) ||
+    equals(type, ResourceType.anomalydetection);
```
Comment (Thebarda) [inline www/front_src/src/Resources/Details/tabs/Graph/index.tsx]: Same comment about the Modal PR

File: www/front_src/src/Resources/Graph/Performance/GraphActions.tsx [hunk 4]
```diff
@@ -64,7 +69,7 @@ const GraphActions = ({
   const [exporting, setExporting] = useState<boolean>(false);
   const { format } = useLocaleDateTimeFormat();
   const navigate = useNavigate();
-
+  const isResourceAD = equals(resourceType, ResourceType.anomalydetection);
```
Comment (bdauria) [inline www/front_src/src/Resources/Graph/Performance/GraphActions.tsx]:
```suggestion
  const isResourceAnomalyDetection = equals(resourceType, ResourceType.anomalydetection);
```

Avoid using shortcuts in naming

File: www/front_src/src/Resources/Graph/Performance/GraphActions.tsx [hunk 5]
```diff
@@ -139,6 +144,18 @@ const GraphActions = ({
           >
             <SaveAsImageIcon style={{ fontSize: 18 }} />
           </IconButton>
+          {isResourceAD && (
```
Comment (bdauria) [inline www/front_src/src/Resources/Graph/Performance/GraphActions.tsx]:
```suggestion
          {isResourceAnomalyDetection && (
```

File: lang/pt_BR.UTF-8/LC_MESSAGES/messages.po [hunk 6]
```diff
@@ -15089,6 +15089,9 @@ msgstr "Erro ao remover tokens de um contato"
 #  msgid "Export to png"
 #  msgstr ""
 
+#  msgid "Edit anomaly detection data"
```
Comment (Thebarda) [inline lang/pt_BR.UTF-8/LC_MESSAGES/messages.po:15092]:
```suggestion
# msgid "Edit anomaly detection data"
```

**Review by Thebarda — COMMENTED** [review]
I think it's better to rename AD to Anomaly Detection because also means Auto Discovery ^^

File: www/front_src/src/Resources/Graph/Performance/GraphActions.tsx [hunk 7]
```diff
@@ -139,6 +147,18 @@ const GraphActions = ({
...
+          {isResourceAnomalyDetection && (
+            <IconButton
+              ariaLabel={t(labelEditAnomalyDetectionData)}
+              data-testid={labelEditAnomalyDetectionData}
+              disabled={isNil(timeline)}
+              size="large"
+              title={t(labelEditAnomalyDetectionData)}
+              onClick={(): void => undefined}
+            >
+              <WrenchIcon style={{ fontSize: 18 }} />
```
Comment (Thebarda) [inline www/front_src/src/Resources/Graph/Performance/GraphActions.tsx]: you can use the `fontSize` props instead

File: www/front_src/src/Resources/Graph/Performance/GraphActions.tsx [hunk 8]
```diff
@@ -139,6 +147,18 @@ const GraphActions = ({
...
             <SaveAsImageIcon style={{ fontSize: 18 }} />
           </IconButton>
+          {isResourceAnomalyDetection && (
+            <IconButton
+              ariaLabel={t(labelEditAnomalyDetectionData)}
+              data-testid={labelEditAnomalyDetectionData}
+              disabled={isNil(timeline)}
+              size="large"
+              title={t(labelEditAnomalyDetectionData)}
+              onClick={(): void => undefined}
```
Comment (Thebarda) [inline www/front_src/src/Resources/Graph/Performance/GraphActions.tsx:176]: Will this be used afterward?

File: www/front_src/src/Resources/Graph/Performance/GraphActions.tsx [hunk 9]
```diff
@@ -64,7 +69,10 @@ const GraphActions = ({
   const [exporting, setExporting] = useState<boolean>(false);
   const { format } = useLocaleDateTimeFormat();
   const navigate = useNavigate();
-
+  const isResourceAnomalyDetection = equals(
```
Comment (Thebarda) [inline www/front_src/src/Resources/Graph/Performance/GraphActions.tsx]:
```suggestion
  const isAnomalyDetectionResource = equals(
```

File: www/front_src/src/Resources/Graph/Performance/GraphActions.tsx [hunk 10]
```diff
@@ -139,6 +147,18 @@ const GraphActions = ({
           >
             <SaveAsImageIcon style={{ fontSize: 18 }} />
           </IconButton>
+          {isResourceAnomalyDetection && (
```
Comment (Thebarda) [inline www/front_src/src/Resources/Graph/Performance/GraphActions.tsx]:
```suggestion
          {isAnomalyDetectionResource && (
```

File: www/front_src/src/Resources/Graph/Performance/GraphActions.tsx (hunk 8 above)
Comment (Noha-ElAbrouki) [reply www/front_src/src/Resources/Graph/Performance/GraphActions.tsx:176]: yess

**Pull request closed without merging**